tests/unit/utils/test_analytics_utils.py

```diff
@@ -11,16 +11,173 @@
+from uuid import UUID
+
+from pytest_mock import MockFixture
 
+from zenml.config.global_config import GlobalConfiguration
+from zenml.enums import StackComponentType
 from zenml.utils.analytics_utils import (
     AnalyticsEvent,
     get_segment_key,
     track_event,
 )
 
 
+class environment_handler(object):
+    """Context manager to manipulate the environment variables during a process."""
+
+    def __init__(self, params: Optional[Dict[str, str]] = None):
+        """Initialization of the manager.
+
+         It saves the provided additional params.
+
+        Args:
+            params: The type of the analytics event
+        """
+        self.original_environment = None
+        self.params = params or {}
+
+    def __enter__(self) -> "environment_handler":
+        """Enter function of the manager.
+
+        It creates a copy of the original set of environment variables and
+        updates it with the provided params.
+
+        Returns:
+            the handler instance.
+        """
+        self.original_environment = os.environ.copy()
+        os.environ.update(self.params)
+
+        return self
+
+    def __exit__(
+        self,
+        type_: Optional[Any],
+        value: Optional[Any],
+        traceback: Optional[Any],
+    ) -> Any:
+        """Exit function of the manager.
+
+        Reverts the environmental variables back to their original state.
+
+        Args:
+            type_: The class of the exception
+            value: The instance of the exception
+            traceback: The traceback of the exception
+
+        """
+        os.environ = self.original_environment
+
+
+def event_check(
+    user_id: str, event: AnalyticsEvent, metadata: Dict[str, str]
+) -> None:
+    """Mock function to replace the 'analytics.track' function for tests.
+
+    By utilizing this mock function, we can validate the inputs given to
+    actual track function.
+
+    Args:
+        user_id: the string representation of the client id
+        event: the type of the event
+        metadata: the collection of metadata describing the event
+
+    Raises:
+        ...
+    """
+    # Check whether the user id corresponds to the client id
+    gc = GlobalConfiguration()
+    assert user_id == str(gc.user_id)
+
+    # Check the validity of the event
+    assert event in AnalyticsEvent.__members__.values()
+
+    # Check the type of each metadata entry
+    assert all([not isinstance(v, UUID) for v in metadata.values()])
+
+    # Check whether common parameters are included in the metadata
+    assert "environment" in metadata
+    assert "python_version" in metadata
+    assert "version" in metadata
+    assert "event_success" in metadata
+
+    if event == AnalyticsEvent.REGISTERED_STACK:
+        assert StackComponentType.ARTIFACT_STORE in metadata
+        assert StackComponentType.ORCHESTRATOR in metadata
+
+        assert "project_id" in metadata
+        assert "entity_id" in metadata
+        assert "user_id" in metadata
```

Comment:
We can also check if the user_id is the same as the user_id in the database?

Reply:
Yes. Fixed this.